src/Microsoft.VisualStudio.Threading/AsyncAutoResetEvent.cs

```diff
@@ -177,7 +177,7 @@ internal WaiterCompletionSource(AsyncAutoResetEvent owner, bool allowInliningCon
                 : base(allowInliningContinuations)
             {
                 this.CancellationToken = cancellationToken;
-                this.Registration = cancellationToken.Register(owner.onCancellationRequestHandler, this);
+                this.Registration = cancellationToken.Register(owner.onCancellationRequestHandler!, this);
```

Comment:
💭 Can we pass in `onCancellationRequestHandler` instead of `owner` so the caller is expected to verify the non-null value?

Reply:
This isn't about `onCancellationRequestHandler` being possibly `null`. It is already typed to guarantee it is non-null. It's because the delegate is typed as `Action<object>` but `Register` only takes `Action<object?>`. I don't care to put the onus on the delegate to check that the argument is non-null, since I guarantee it won't be because I pass in a state object here.

Reply:
I wrapped this in a helper method as you suggested.